### PR TITLE
mongo-c-driver 1.17.0

### PR DESCRIPTION
**Pull request by kevinAlbs**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/mongo-c-driver.rb [hunk 1]
```diff
@@ -21,8 +21,11 @@ class MongoCDriver < Formula
 
   def install
     cmake_args = std_cmake_args
-    cmake_args << "-DBUILD_VERSION=1.17.0-pre" if build.head?
+    cmake_args << "-DBUILD_VERSION=1.18.0-pre" if build.head?
     system "cmake", ".", *cmake_args
+    # Replace full stored compiler path.
+    cc_path = `type -path $CC`
+    inreplace "src/libmongoc/src/mongoc/mongoc-config.h", cc_path.chomp, ENV.cc
```
Comment (kevinAlbs) [inline Formula/mongo-c-driver.rb:28]: libmongoc stores the compiler in the generated mongoc-config.h. This replaces the full path with just the compiler name.

Similar to the issue described here: https://discourse.brew.sh/t/updating-root-formula-a-few-questions/8099

Comment (SeekingMeaning) [inline Formula/mongo-c-driver.rb:28]:
```suggestion
    inreplace "src/libmongoc/src/mongoc/mongoc-config.h", "@MONGOC_CC@", ENV.cc
    system "cmake", ".", *cmake_args
```

Comment (kevinAlbs) [reply Formula/mongo-c-driver.rb]: Good thinking, thanks!

**Review by SeekingMeaning — APPROVED** [review]
Thanks again for your contribution to Homebrew/core, @kevinAlbs! 🌟

**Pull request closed without merging**